federation/service/service.go

```diff
@@ -38,6 +40,8 @@ var (
 
 	DefaultActionlog actionlog.Recorder
 
+	DefaultNode NodeService
+
 	// wrapper around time.Now() that will aid service testing
 	now = func() time.Time {
```

Comment:
I suggest you change rval to *time.Time, it's easier to convert to time.Time if needed then other way around...